api/v1alpha2/iut_types.go

```diff
@@ -0,0 +1,107 @@
+// Copyright Axis Communications AB.
+//
+// For a full list of individual contributors, please see the commit history.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha2
+
+import (
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// IutSpec defines the desired state of Iut
+type IutSpec struct {
+	// ID is the ID for the IUT. The ID is a UUID, any version, and regex matches that.
+	// +kubebuilder:validation:Pattern="^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$"
+	// +required
+	ID string `json:"id"`
+
+	// Identity is the PackageURL definition of the IUT.
+	// +required
+	Identity string `json:"identity"`
+
+	// EnvironmentRequest is the name of the environmentrequest which requested this IUT.
+	// +required
+	EnvironmentRequest string `json:"environmentRequest"`
+
+	// ProviderID is the name of the Provider used to create this Iut.
+	// +required
+	ProviderID string `json:"provider_id"`
+
+	// ProviderData is specific data provided by the IUT providers
+	// +optional
+	ProviderData *apiextensionsv1.JSON `json:"provider_data,omitempty"`
+}
+
+// IutStatus defines the observed state of Iut.
+type IutStatus struct {
+	// For Kubernetes API conventions, see:
+	// https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties
+
+	// conditions represent the current state of the Iut resource.
+	// Each condition has a unique type and reflects the status of a specific aspect of the resource.
```

Comment:
I don't understand what these aspects are?  Are we referring to Available, Progressing and Degraded?